File: schemas/recipe-cooking/drizzle/recipe_images.ts
// recipe_images: Photo attachments for recipes with ordering and primary flag.
// See README.md for full design rationale.

import { pgTable, uuid, text, integer, boolean, timestamp, index } from "drizzle-orm/pg-core";
import { recipes } from "./recipes";

export const recipeImages = pgTable(
  "recipe_images",
  {
    id: uuid("id").primaryKey().defaultRandom(),
    recipeId: uuid("recipe_id").notNull().references(() => recipes.id, { onDelete: "cascade" }),
    imageUrl: text("image_url").notNull(),
    caption: text("caption"),
    isPrimary: boolean("is_primary").notNull().default(false),
    position: integer("position").notNull().default(0),
    createdAt: timestamp("created_at", { withTimezone: true }).notNull().defaultNow(),
  },
  (table) => [
    index("idx_recipe_images_recipe_id_position").on(table.recipeId, table.position),
  ]
);

File: schemas/recipe-cooking/spacetimedb/recipe_images.rs
// recipe_images: Photos and images associated with a recipe.
// See README.md for full design rationale.

use spacetimedb::Timestamp;

#[spacetimedb::table(name = recipe_images, public)]
pub struct RecipeImage {
    #[primary_key]
    pub id: String, // UUID

    #[index(btree)]
    pub recipe_id: String, // FK → recipes.id (cascade delete)

    pub image_url: String,
    pub caption: Option<String>,
    pub is_primary: bool, // default false
    pub position: i32, // default 0
    pub created_at: Timestamp,
}
